parachain/modules/ismp/pallet/src/mmr.rs

```diff
@@ -15,6 +15,16 @@
 
 // This module contains code adapted from https://github.com/paritytech/substrate/blob/master/frame/merkle-mountain-range/src/mmr/mod.rs
 
+/// <HB SBP Review
+///
+/// disclaimer: This is a general comment without having the full context,
+/// but i wonder if the MMR FRAME pallet could be used and wrap the specific logic needed
+/// instead of re-implementing it and adapting the code.
+///
+/// https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/merkle-mountain-range
+///
+/// (beefy and mmr is not my area of expertise)
+/// >
```

Comment:
Unfortunately not, we only want to finalize once per block. The MMR pallet finalizes per leaf inserted, which isn't very performant